naga/src/proc/constant_evaluator.rs

```diff
@@ -2497,6 +2520,121 @@ impl<'a> ConstantEvaluator<'a> {
 
         Ok(resolution)
     }
+
+    fn select(
+        &mut self,
+        reject: Handle<Expression>,
+        accept: Handle<Expression>,
+        condition: Handle<Expression>,
+        span: Span,
+    ) -> Result<Handle<Expression>, ConstantEvaluatorError> {
+        let arg = |this: &mut Self, expr| {
+            let expr = this.check_and_get(expr)?;
+            let expr = this.eval_zero_value_and_splat(expr, span)?;
+            let expr: &Expression = &this.expressions[expr];
+            this.try_eval_and_append(expr.clone(), span)
```

Comment:
It shouldn't be necessary to call `try_eval_and_append` here. Since `check_and_get` didn't fail, that means the `Select`'s operands are known to be constant values, which means they must have been created by `try_eval_and_append` previously.

Note that we don't see these sorts of calls to `try_eval_and_append` elsewhere in the module.

Reply:
I actually wasn't sure if we were calling constant evaluation in a bottom-up or top-down direction. It sounds like we're calling this bottom-up, then, which is great!

Reply:
Just confirmed that both `check_and_get` _and_ `try_eval_and_append` here weren't necessary; force-pushed f03f9bd65942.